### DISCUSSION TITLE
How to place grafana dashboard in a folder other than the default folder

### DISCUSSION THREAD

**Discussion by perlinger** [prometheus-operator/kube-prometheus#1452] [Q&A] [answered]
My prom.jsonnet is shown below. We created kubeflow-dashboard.json and argocd-dashboard.json dashboards.
The code below places our dashboards in grafana's default folder. How would we place our dashboards in another grafana folder?

```
local datasource = import './datasources.libsonnet';
local kp = 
(import 'kube-prometheus/kube-prometheus.libsonnet') +
(import 'kube-prometheus/kube-prometheus-eks.libsonnet') +
(import "prometheus-pushgateway/pushgateway.libsonnet") +
(import 'kube-prometheus/kube-prometheus-custom-metrics.libsonnet') + 
{
  _config+:: {
    namespace: 'monitoring',
    grafana+:: {
      datasources: datasource,
      dashboards+:: {
        'kubeflow-dashboard.json': (import './kubeflow-dashboard.json'),
        'argocd-dashboard.json': (import './argocd-dashboard.json'),
      },
    }
  },
};
{ ['setup/0namespace-' + name]: kp.kubePrometheus[name] for name in std.objectFields(kp.kubePrometheus) } +
{
  ['setup/prometheus-operator-' + name]: kp.prometheusOperator[name]
  for name in std.filter((function(name) name != 'serviceMonitor'), std.objectFields(kp.prometheusOperator))
} +
// serviceMonitor is separated so that it can be created after the CRDs are ready
{ 'prometheus-operator-serviceMonitor': kp.prometheusOperator.serviceMonitor } +
{ ['pushgateway-' + name]: kp.pushgateway[name] for name in std.objectFields(kp.pushgateway) } +
{ ['node-exporter-' + name]: kp.nodeExporter[name] for name in std.objectFields(kp.nodeExporter) } +
{ ['kube-state-metrics-' + name]: kp.kubeStateMetrics[name] for name in std.objectFields(kp.kubeStateMetrics) } +
{ ['prometheus-' + name]: kp.prometheus[name] for name in std.objectFields(kp.prometheus) } +
{ ['prometheus-adapter-' + name]: kp.prometheusAdapter[name] for name in std.objectFields(kp.prometheusAdapter) } +
{ ['alertmanager-' + name]: kp.alertmanager[name] for name in std.objectFields(kp.alertmanager) } +
{ ['grafana-' + name]: kp.grafana[name] for name in std.objectFields(kp.grafana) }
```

**Comment by perlinger** [comment]
This works...No issue...
'''
local datasource = import './datasources.libsonnet';
local kp = 
(import 'kube-prometheus/kube-prometheus.libsonnet') +
(import 'kube-prometheus/kube-prometheus-eks.libsonnet') +
(import "prometheus-pushgateway/pushgateway.libsonnet") +
(import 'kube-prometheus/kube-prometheus-custom-metrics.libsonnet') + 
{
  _config+:: {
    namespace: 'monitoring',
    grafana+:: {
      datasources: datasource,
      folderDashboards+:: {
        KubeFlow_Federated:{
          'kubeflow-dashboard.json': (import './kubeflow-dashboard.json'),
        },
        Arcocd: {
          'argocd-dashboard.json': (import './argocd-dashboard.json'),
        },
      },
    }
  },
};
{ ['setup/0namespace-' + name]: kp.kubePrometheus[name] for name in std.objectFields(kp.kubePrometheus) } +
{
  ['setup/prometheus-operator-' + name]: kp.prometheusOperator[name]
  for name in std.filter((function(name) name != 'serviceMonitor'), std.objectFields(kp.prometheusOperator))
} +
// serviceMonitor is separated so that it can be created after the CRDs are ready
{ 'prometheus-operator-serviceMonitor': kp.prometheusOperator.serviceMonitor } +
{ ['pushgateway-' + name]: kp.pushgateway[name] for name in std.objectFields(kp.pushgateway) } +
{ ['node-exporter-' + name]: kp.nodeExporter[name] for name in std.objectFields(kp.nodeExporter) } +
{ ['kube-state-metrics-' + name]: kp.kubeStateMetrics[name] for name in std.objectFields(kp.kubeStateMetrics) } +
{ ['prometheus-' + name]: kp.prometheus[name] for name in std.objectFields(kp.prometheus) } +
{ ['prometheus-adapter-' + name]: kp.prometheusAdapter[name] for name in std.objectFields(kp.prometheusAdapter) } +
{ ['alertmanager-' + name]: kp.alertmanager[name] for name in std.objectFields(kp.alertmanager) } +
{ ['grafana-' + name]: kp.grafana[name] for name in std.objectFields(kp.grafana) }
```

**Comment by AlHood77** [comment, marked as answer]
This is how I did it

```jsonnet
    values+:: {
      common+: {
        namespace: 'monitoring',
      },
      prometheus+: {
        namespaces+: [],
      },
      grafana+:: {
        folderDashboards+:: {
          Postgres: {
            'postgres.json': (import '../dashboards/postgres.json'),
          },
        },
      },
    },
```
